### ISSUE TITLE
export * as namespace does not support string literals

### ISSUE THREAD

**Issue by dnalborczyk** [rollup/rollup#4784]
I believe this is part of `arbitrary module namespace names`: https://github.com/rollup/rollup/issues/4322

### Rollup Version

v3.9.1

### Operating System (or Browser)

n/a

### Node Version (if applicable)

n/a

### Link To Reproduction

https://rollupjs.org/repl/?version=3.9.1&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMColMjBhcyUyMCU1QyUyMmZvbyU1QyUyMiUyMGZyb20lMjAlNUMlMjIuJTJGZm9vLmpzJTVDJTIyJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIyZm9vLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMGNvbnN0JTIwZm9vJTIwJTNEJTIwJ2JhciclMjIlMkMlMjJpc0VudHJ5JTIyJTNBZmFsc2UlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyZXMlMjIlMkMlMjJuYW1lJTIyJTNBJTIybXlCdW5kbGUlMjIlMkMlMjJhbWQlMjIlM0ElN0IlMjJpZCUyMiUzQSUyMiUyMiU3RCUyQyUyMmdsb2JhbHMlMjIlM0ElN0IlN0QlN0QlMkMlMjJleGFtcGxlJTIyJTNBbnVsbCU3RA==

### Expected Behaviour

```js
export * as "foo" from "./foo.js"
```

### Actual Behaviour

SyntaxError

**Comment by lukastaegert** [comment]
Ah, I overlooked you can do this with namespaces as well. The fix is probably very similar to what I did in #4770, if you want, you can have a shot at fixing this.

**Comment by lukastaegert** [comment]
Actually forget what I said, I have overlooked it but if it is a Syntax error, then the acorn parser has also overlooked it and it needs to be fixed in acorn. Until then, there is little we can do about it in Rollup.

**Comment by dnalborczyk** [comment]
just tried `acorn` and seems to be working:

```js
export * as "foo" from "foo"
```

**acorn**
```json
{
  "type": "Program",
  "start": 0,
  "end": 28,
  "body": [
    {
      "type": "ExportAllDeclaration",
      "start": 0,
      "end": 28,
      "exported": {
        "type": "Literal",
        "start": 12,
        "end": 17,
        "value": "foo",
        "raw": "\"foo\""
      },
      "source": {
        "type": "Literal",
        "start": 23,
        "end": 28,
        "value": "foo",
        "raw": "\"foo\""
      }
    }
  ],
  "sourceType": "module"
}
```

**rollup**
```shell
roll.js → bundle.js...
[!] RollupError: Unexpected token
roll.js (1:12)
1: export * as "foo" from "foo"
               ^
    at error (/node_module/rollup/dist/shared/rollup.js:210:30)
    at Module.error (/node_modules/rollup/dist/shared/rollup.js:13478:16)
    at Module.tryParse (/node_modules/rollup/dist/shared/rollup.js:13872:25)
    at Module.setSource (/node_modules/rollup/dist/shared/rollup.js:13763:39)
    at ModuleLoader.addModuleSource (/node_modules/rollup/dist/shared/rollup.js:23404:20)
```

**Comment by lukastaegert** [comment]
Maybe we are not using the latest acorn version? I mean, tryParse from the stack trace is doing nothing but calling acorn.

**Comment by dnalborczyk** [comment]
good point. debugged it a bit, and I'm fairly sure the parser error is being thrown by the `import assertions` plugin 😞

**Comment by sapphi-red** [comment]
The error seems to be changed since 4.0.0.
https://rollupjs.org/repl/?version=4.0.2&shareable=JTdCJTIyZXhhbXBsZSUyMiUzQW51bGwlMkMlMjJtb2R1bGVzJTIyJTNBJTVCJTdCJTIybmFtZSUyMiUzQSUyMm1haW4uanMlMjIlMkMlMjJjb2RlJTIyJTNBJTIyZXhwb3J0JTIwKiUyMGFzJTIwJTVDJTIyZm9vJTVDJTIyJTIwZnJvbSUyMCU1QyUyMi4lMkZmb28uanMlNUMlMjIlMjIlMkMlMjJpc0VudHJ5JTIyJTNBdHJ1ZSU3RCUyQyU3QiUyMm5hbWUlMjIlM0ElMjJmb28uanMlMjIlMkMlMjJjb2RlJTIyJTNBJTIyZXhwb3J0JTIwY29uc3QlMjBmb28lMjAlM0QlMjAnYmFyJyUyMiUyQyUyMmlzRW50cnklMjIlM0FmYWxzZSU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlN0QlN0Q=

The error points this line.
https://github.com/rollup/rollup/blob/d90f274832b28429bd593ad0e34791ca7c2888ba/src/Module.ts#L572

**Comment by lukastaegert** [comment]
Fix at #5298.
